functions/index.js

```diff
@@ -124,9 +137,59 @@ const loginUser = (req, res) => {
 };
 
 
+const logoutUser = (req, res) => {
+    // To do, this doesn't work at the moment, needs fixed.
+    firebase.auth().signOut().then(function() {
+        console.log('Signed out');
+    })
+    .catch(function(error) {
+        console.error('Sign out error', error);
+    });
+}
+
+const user = firebase.auth().currentUser;
+    user.updateProfile({
+    displayName: "prueba",
+}).then(() => {
+  console.log('successful')
+}).catch((error) => {
+    console.log('error')
+});
+
+const updateUser = firebase.auth().currentUser;
+user.updateProfile({
+  displayName: "backend",
+  photoURL: "https://example.com/jane-q-user/profile.jpg"
+}).then(() => {
+  // Update successful
+  // ...
+}).catch((error) => {
+  // An error occurred
+  // ...
+});  
+
+const deleteUser = firebase.auth().currentUser;
+
+user.delete().then(() => {
+  // User deleted.
+}).catch((error) => {
+  // An error ocurred
+  // ...
+});
+
+
+
+
+
+
+
 // User routes
 app.post("/user", signupUser);
 app.post("/user/login", loginUser);
+app.post("/user/logout", logoutUser);
+app.post("/user/ {username}", username);
+app.post("/user/ {username}", updateUser);
+app.post("/user/ {username}", deleteUser);
```

Comment:
A couple of things here:
* You must not specify a space in the API, because in URLS we don't normally put spaces, and because it's not defined like that: ![image](https://user-images.githubusercontent.com/7826376/163128900-7e62a327-ed00-42e8-acc0-7d26e5459a01.png)
* The endpoint is exactly the same, so it's unclear what will happen when we call the API in `/user/{username}` (which function will be called? username, updateUser, deleteUser?_